### PR TITLE
Workaround change to DECTRIS file writer

### PR DESCRIPTION
**Pull request by graeme-winter**
Now look for bit_depth_image which appears to be documented in their standard file description, despite not being NXmx compliant. This is a local fix to the file.

Fixes dials/dials#2401

### CODE REVIEW

**Comment by graeme-winter** [comment]
This one _does not_ depend on an upstream fix

**Comment by dperl-dls** [comment]
Will this mean that we should include both `bit_depth_readout` and `bit_depth_image` in the nxs files we write, and that the correct value will be used for checks in that case?

**Comment by graeme-winter** [comment]
> Will this mean that we should include both `bit_depth_readout` and `bit_depth_image` in the nxs files we write, and that the correct value will be used for checks in that case?

In the ever-so-slightly longer term absolutely yes, I think this is the right thing to do (once the dust has settled and we have confirmation that `bit_depth_image` is a thing)

**Comment by graeme-winter** [comment]
https://github.com/nexusformat/definitions/issues/1258

**Comment by rjgildea** [comment]
> Will this mean that we should include both `bit_depth_readout` and `bit_depth_image` in the nxs files we write, and that the correct value will be used for checks in that case?

This changeset only applies to files written by the DECTRIS eiger filewriter, i.e. they don't apply to DLS-flavour NeXus files which aim to follow official NeXus/NXmx standard (`bit_depth_image` isn't currently part of the standard - perhaps it should be, but that's a separate question).

**Comment by graeme-winter** [comment]
> > Will this mean that we should include both `bit_depth_readout` and `bit_depth_image` in the nxs files we write, and that the correct value will be used for checks in that case?
> 
> This changeset only applies to files written by the DECTRIS eiger filewriter, i.e. they don't apply to DLS-flavour NeXus files which aim to follow official NeXus/NXmx standard (`bit_depth_image` isn't currently part of the standard - perhaps it should be, but that's a [separate question](https://github.com/nexusformat/definitions/issues/1258)).

True, but the values _are_ in the header packet and _are_ written in and literally today there are some routes where we now point `bit_depth_readout` in our NXmx files to `bit_depth_image` which ... feels wrong. Having both would seem sensible.

File: src/dxtbx/format/FormatNXmxEigerFilewriter.py [hunk 1]
```diff
@@ -28,6 +29,19 @@ def __init__(self, image_file, **kwargs):
         """Initialise the image structure from the given file."""
         super().__init__(image_file, **kwargs)
 
+    def _start(self):
+        super()._start()
+        try:
+            # This is (currently) a DECTRIS-specific non-standard item that
+            # we will use in preference to bit_depth_readout (see below)
+            self._bit_depth_image = int(
+                self._cached_file_handle["entry/instrument/detector/bit_depth_image"][
```
Comment (ndevenish) [inline src/dxtbx/format/FormatNXmxEigerFilewriter.py:38]: This works without a leading `/` ?

File: src/dxtbx/format/FormatNXmxEigerFilewriter.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 from __future__ import annotations
 
 import re
+from typing import Optional
```
Comment (ndevenish) [inline src/dxtbx/format/FormatNXmxEigerFilewriter.py:4]:
```suggestion
```

File: src/dxtbx/format/FormatNXmxEigerFilewriter.py [hunk 3]
```diff
@@ -63,7 +84,10 @@ def get_raw_data(self, index):
 
 
 def get_raw_data(
-    nxdata: nxmx.NXdata, nxdetector: nxmx.NXdetector, index: int
+    nxdata: nxmx.NXdata,
+    nxdetector: nxmx.NXdetector,
+    index: int,
+    bit_depth: Optional[int] = None,
```
Comment (ndevenish) [inline src/dxtbx/format/FormatNXmxEigerFilewriter.py:90]:
```suggestion
    bit_depth: int | None = None,
```
We have `__future__.annotations` turned on here

**Review by ndevenish — APPROVED** [review]

File: src/dxtbx/format/FormatNXmxEigerFilewriter.py (hunk 1 above)
Comment (graeme-winter) [reply src/dxtbx/format/FormatNXmxEigerFilewriter.py:37]: Yes, it's a relative path then

Comment (graeme-winter) [reply src/dxtbx/format/FormatNXmxEigerFilewriter.py:37]: But yeh, could prefix with `/` equally

**Pull request merged**